br/pkg/lightning/backend/local/duplicate.go

```diff
@@ -309,10 +374,62 @@ func (manager *DuplicateManager) CollectDuplicateRowsFromLocalIndex(
 		SQLMode: mysql.ModeStrictAllTables,
 	})
 	if err != nil {
-		return err
+		return errors.Trace(err)
 	}
-	handles := make([][]byte, 0)
+
 	allRanges := make([]tidbkv.KeyRange, 0)
+	{
+		// Collect row handle duplicates.
+		var dataConflictInfos []errormanager.DataConflictInfo
+		ranges := ranger.FullIntRange(false)
+		if tbl.Meta().IsCommonHandle {
+			ranges = ranger.FullRange()
+		}
+		keyRanges, err := distsql.TableHandleRangesToKVRanges(nil, []int64{tbl.Meta().ID}, tbl.Meta().IsCommonHandle, ranges, nil)
```

Comment:
I'm not sure whether we shall use `TableHandleRangesToKVRanges` instead of `TableRangesToKVRanges`. If this method is correct, we shall replace all `TableRangesToKVRanges` of BR with `TableHandleRangesToKVRanges`.

Reply:
the comment on `TableRangesToKVRanges` says (lines 358-359)

https://github.com/pingcap/tidb/blob/db4a34ee84994150aaab529f7e658cbe286d5bee/distsql/request_builder.go#L357-L362

so i think yes. 

currently `TableRangesToKVRanges(tid, ranges, fb)` is equivalent to `TableHandleRangesToKVRanges(nil, []int64{tid}, false, ranges, fb)`.